lib/ansible/plugins/connection/winrm.py

```diff
@@ -21,11 +26,55 @@
             - name: ansible_host
             - name: ansible_winrm_host
       remote_user:
+        keywords:
+          - name: user
+          - name: remote_user
         description:
             - The user to log in as to the Windows machine
         vars:
             - name: ansible_user
             - name: ansible_winrm_user
+      port:
+        description: port for winrm to connect on remote target
+        vars:
+          - name: ansible_port
+          - name: ansible_winrm_port
+        default: 5985
+        keywords:
+          - name: port
+        type: integer
+      scheme:
+        description:
+            - URI scheme to use
+        choices: [http, https]
+        default: http
+        vars:
+          - name ansible_winrm_scheme
+      path:
+        description: URI path to connect to
+        default: '/wsman'
+        vars:
+          - name: ansible_winrm_path
+      transport:
+        description:
+           - List of winrm transports to attempt to to use (ssl, plaintext, kerberos, etc)
+           - If None (the default) the plugin will try to automatically guess the correct list
+           - The choices avialable depend on your version of pywinrm
```

Comment:
Might want to add the latest options since pywinrm `0.2.2`, these are
* basic
* certificate
* ntlm
* kerberos
* credssp

There is also `plaintext` and `ssl` but they are an amalgamation of transport and scheme which i don't think we want to encourage. Thoughts @nitzmahone?

Reply:
Yeah, I'd go with that list- plaintext and ssl are legacy (technically `basic` is an alias for `plaintext` and `certificate` is "sort of" an alias to `ssl`).